src/coffea/analysis_tools.py

```diff
@@ -593,8 +1147,120 @@ def any(self, *names):
         returns a boolean array where an entry is True if the corresponding entries
         ``cut1 == True`` or ``cut2 == False``, and ``cut3`` arbitrary.
         """
+        for cut in names:
+            if not isinstance(cut, str) or cut not in self._names:
+                raise ValueError(
+                    "All arguments must be strings that refer to the names of existing selections"
+                )
         consider = 0
         for name in names:
             idx = self._names.index(name)
             consider |= 1 << idx
-        return (self._data & consider) != 0
+        return (self._data & self._dtype.type(consider)) != 0
+
+    def nminusone(self, *names):
+        """Compute the "N-1" style selection for a set of selections
+
+        The N-1 style selection for a set of selections, returns an object which can return a list of the number of events
+        that pass all the other selections ignoring one at a time. The first element of the returned list
+        is the total number of events before any selections are applied.
+        The last element is the final number of events that pass if all selections are applied.
+        It also returns a list of boolean mask vectors of which events pass the N-1 selection each time.
+        Can also return a histogram as a ``hist.Hist`` object where the bin heights are the number of events of the N-1 selection list.
+        If the PackedSelection is in delayed mode, the elements of those lists will be dask_awkward Arrays that can be computed whenever the user wants.
+        If the histogram is requested, the delayed arrays of the number of events list will be computed in the process in order to set the bin heights.
+
+        Parameters
+        ----------
+            ``*names`` : args
+                The named selections to use, need to be a subset of the selections already added
+
+        Returns
+        -------
+            res: coffea.analysis_tools.NminusOne
+                A wrapper class for the results, see the documentation for that class for more details
+        """
+        for cut in names:
+            if not isinstance(cut, str) or cut not in self._names:
+                raise ValueError(
+                    "All arguments must be strings that refer to the names of existing selections"
+                )
+
+        masks = []
+        for i, cut in enumerate(names):
+            mask = self.all(*(names[:i] + names[i + 1 :]))
+            masks.append(mask)
+        mask = self.all(*names)
+        masks.append(mask)
+
+        if not self.delayed_mode:
+            nev = [len(self._data)]
+            nev.extend(numpy.sum(masks, axis=1))
+
+        else:
+            nev = [
+                dask_awkward.sum(
+                    dask_awkward.from_awkward(awkward.Array([len(self._data)]), 1)
```

Comment:
this will be very inefficient at scale (unless the partitions are precomputed), couldn't this be expressed in delayed way?

Reply:
`dak.count(self._data, axis=0)` seems like it should do, no?

Reply:
I'll try to fix this one. I'm now fixing the above comments and changing the `to_npz` to mimmick the `to_parquet` behavior